### PR TITLE
fix #662 repeated loading DataSourceManager

### LINKED ISSUE #662: Failed at async committing ... null
- [ ] I have searched the [issues](https://github.com/alibaba/fescar/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
When TC send branchCommit request to RM, RM log "Failed at async committing ... null"

### Ⅱ. Describe what happened

AsyncWorker failed to do branchCommit, and left undolog.

### Ⅲ. Describe what you expected to happen

AsyncWorker deletes undolog.

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

I find there are two way to instantiate the Class DataSourceManager
1. DefaultResourceManager uses EnhancedServiceLoader to instantiate DataSourceManager 
 EnhancedServiceLoader call DataSourceManager() and DataSourceManager.init(). Then register DataSourceProxy to this DataSourceManager.

2. DataSourceManager's member AsyncWorker call DataSourceManager.get() to instantiate  another DataSourceManager.

3. AbstractRMHandler uses the first DataSourceManager instantiation to do branchCommit.

4. The first DataSourceManager pass BranchCommit to its AsyncWorker.

5. While AsyncWorker uses the second DataSourceManager instantiation to get DataSourceProxy ,which no DataSourceProxy is registered.

6. AsyncWorker.doBranchCommit throw NPE at line158.

### Ⅵ. Environment:

- JDK version : JDK 1.8
- OS : linux
- Others: Fescar Version 0.4.0

### PR DESCRIPTION
**Pull request by github-ygy**
### Ⅰ. Describe what this PR did
repeated loading DataSourceManager, we need to unify the DataSourcemanager instance. 

### Ⅱ. Does this pull request fix one issue?

fix #662 #667

### CODE REVIEW

File: rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/AsyncWorker.java [hunk 1]
```diff
@@ -16,30 +16,26 @@
...
 import com.alibaba.fescar.core.model.ResourceManagerInbound;
+import com.alibaba.fescar.rm.DefaultResourceManager;
 import com.alibaba.fescar.rm.datasource.undo.UndoLogManager;
-
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.*;
```
Comment (xingfudeshi) [inline rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/AsyncWorker.java]: import *?

Comment (github-ygy) [reply rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/AsyncWorker.java]: idea do it auto

File: test/src/main/java/com/alibaba/fescar/test/client/AppTest.java [hunk 2]
```diff
@@ -1,191 +0,0 @@
...
-    private static final String TX_SERVICE_GROUP = "my_test_tx_group";
-
-    /**
-     * The entry point of application.
-     *
-     * @param args the input arguments
-     */
-    public static void main(String[] args) {
-        TMClient.init(APPLICATION_ID, TX_SERVICE_GROUP);
-        RMClientAT.init(APPLICATION_ID, TX_SERVICE_GROUP);
```
Comment (slievrly) [inline test/src/main/java/com/alibaba/fescar/test/client/AppTest.java]: need delete？ RMClientAT -> RMClient

Comment (github-ygy) [reply test/src/main/java/com/alibaba/fescar/test/client/AppTest.java]: done

**Pull request closed without merging**